core/storage/base_model/gae_models.py

```diff
@@ -663,7 +663,7 @@ class VersionedModel(BaseModel):
     # version number starts at 1 when the model instance is first created.
```

Comment:
Just a question about this. If the version number starts at 1, then why is the default 0? I also don't recall seeing any models with version 0 in the datastore so I just wanted to double-confirm that 0 is the correct thing to go for here and elsewhere.

Reply:
I looked into it and foud the following:
We initialize an entity with version 0 (while creating a domain object of entity) and while committing the model we increase the version number by 1 so all the newly created models have version number 1.

Few references:
1. ([Link](https://github.com/oppia/oppia/pull/11427/files#diff-d3c2dff20265bf6ae06f0aa5e15805b81baf1b75d5822ddc13ff74374d0172b7L921)) All the domain object for the default entity 

2. ([Link](https://github.com/oppia/oppia/pull/11427/files#diff-264272c88885bc2eb7dfbe1d328056b66ca898b19eb8f3c5855718fd9b1b1ccdL951)) We use version 0 to indicate a model is newly created.

3. ([Link](https://github.com/oppia/oppia/pull/11427/files#diff-264272c88885bc2eb7dfbe1d328056b66ca898b19eb8f3c5855718fd9b1b1ccdL762)) When committing a versioned model the version number is increased by 1.

> (Please also explain how you verified that the default of 0 is correct and why it makes sense conceptually.)

Conceptually I'm not sure why we always increase the version number I was assuming that the commit method to handle the version number and _trusted_commit should only commit the changes. (not sure!)

Reply:
Thanks for checking. This seems reasonable but then the "version" field in the storage model should not have any default value specified, right? (Though it should be required.)